### DISCUSSION TITLE
SQL language server not starting

### DISCUSSION THREAD

**Discussion by RizaHKhan** [LunarVim/LunarVim#4210] [Q&A] [answered]
So I'm trying to get [sqlls](https://github.com/joe-re/sql-language-server) to work on `sql` files but its being difficult.

I did: `Lspinstall sqlls`. 

And confirmed it is installed.
![image](https://github.com/LunarVim/LunarVim/assets/51720003/aa098c34-a82a-469a-9754-9abcb6040ca3)

But it will not start:
![image](https://github.com/LunarVim/LunarVim/assets/51720003/a4e63b42-4d80-4b16-b9a7-af4d7db72b4b)

Any idea how to debug?

Thank you,

**Comment by LostInTheLogs** [comment]
see :LspInfo

**Reply by RizaHKhan** [reply]
@LostNeophyte 
<img width="701" alt="image" src="https://github.com/LunarVim/LunarVim/assets/51720003/896f27b3-9205-4e2b-9ecb-a1fc0614987a">

There is no configured list, does this mean I have to do something special in my config file?

**Reply by LostInTheLogs** [reply, marked as answer]
sqlls is disabled by default, remove it from skipped_list, there's an example [here](https://www.lunarvim.org/docs/master/configuration/language-features/language-servers#changing-the-default-server)

**Comment by eboody** [comment]
TL;DR: To enable sqlls (after installing it) in LunarVim and set the current working directory as the root directory, add these configurations to your config.lua:

```lua
lvim.lsp.automatic_configuration.skipped_servers = vim.tbl_filter(function(server)
  return server ~= "sqlls"
end, lvim.lsp.automatic_configuration.skipped_servers)

require("lvim.lsp.manager").setup("sqlls", {
  cmd = {"sql-language-server", "up", "--method", "stdio"};
  filetypes = {"sql", "mysql"};
  root_dir = function() return vim.loop.cwd() end;
})
```

After making changes to your config.lua, don't forget to run `:LvimCacheReset` in LunarVim for the changes to take effect.

Explanation:

The sqlls language server may fail to start if it can't determine a root directory for your project. SQL files don't have an inherent concept of a "project root", which can lead to this issue.

In the code above, we are:

Enabling sqlls: We remove sqlls from the skipped_servers list in LunarVim, effectively enabling the server for use.

```lua
lvim.lsp.automatic_configuration.skipped_servers = vim.tbl_filter(function(server)
  return server ~= "sqlls"
end, lvim.lsp.automatic_configuration.skipped_servers)
```

Setting up sqlls: We configure sqlls using LunarVim's LSP manager, specifying the command to start the server, the file types it should be used with, and setting the root directory to the current working directory.

```lua
require("lvim.lsp.manager").setup("sqlls", {
  cmd = {"sql-language-server", "up", "--method", "stdio"};
  filetypes = {"sql", "mysql"};
  root_dir = function() return vim.loop.cwd() end;
})
```

Using LunarVim's LSP manager to setup the language server ensures that the server is not configured twice, which could happen if we used lspconfig directly after removing the server from skipped_servers.

**Reply by LostInTheLogs** [reply]
if you're removing the lsp from skipped_servers don't configure it manually, it will be configured two times
the best way to do this is 
```lua
require("lvim.lsp.manager").setup("sqlls", {
  cmd = {"sql-language-server", "up", "--method", "stdio"};
  filetypes = {"sql", "mysql"};
  root_dir = function() return vim.loop.cwd() end;
})
```
in .config/lvim/ftplugin.sql.lua

nothing more needed

**Reply by eboody** [reply]
Thanks for this! I edited my comment.

Simply removing sqlls from the skipped list wasn't sufficient for me. 

When I ran :LspInfo it said that the root_directory was not found.

**Reply by dbriemann** [reply]
For me it worked when putting the snippet from @LostNeophyte into `$LUNARVIM_CONFIG_DIR/after/ftplugin/sql.lua`. No need to alter the `skipped_servers`.
